gen/wrap.jl

```diff
@@ -0,0 +1,187 @@
+#=
+Use Clang.jl to wrap the GDAL C API.
+
+Several custom transformations are applied that should make using this package more convenient.
+
+- docstrings are added, created from GDAL Doxygen XML output
+- function names are lowercased following Julia conventions
+- functions that return a Cstring are wrapped in unsafe_string to return a String
+- functions that return a Ptr{Cstring} are wrapped in unsafe_loadstringlist to return a Vector{String}
+- functions that return a Ptr{Cvoid} are wrapped in failsafe to catch errors
+=#
+
+using Clang
+using MacroTools
+using Logging
+using EzXML
+using CEnum  # to be able to include common.jl
+
+const output_dir = joinpath(@__DIR__, "..", "src")
+const xmlpath = joinpath(@__DIR__, "doxygen.xml")
+
+# used to eval if a return type is Ptr{Cvoid}
+include(joinpath(@__DIR__, "..", "src", "common.jl"))
+
+# set up a global logger to log.txt to store the large amount of logging
+loghandle = open(joinpath(@__DIR__, "log.txt"), "w")
+logger = SimpleLogger(loghandle)
+global_logger(logger)
+
+# several functions for building docstrings
+include(joinpath(@__DIR__, "doc.jl"))
+
+# should be here if you pkg> build GDAL
+includedir = normpath(joinpath(@__DIR__, "..", "deps", "usr", "include"))
```

Comment:
can we throw an error with the comment "should be here if you pkg> build GDAL" as a message if this line fails?